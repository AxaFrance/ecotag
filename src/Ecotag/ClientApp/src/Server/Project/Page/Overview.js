import React from 'react';
import {
    ArticleRestitution,
    HeaderRestitution,
    Restitution,
    SectionRestitution,
    SectionRestitutionColumn,
    SectionRestitutionRow
} from '@axa-fr/react-toolkit-restitution';
import {formatTimestampToString} from "../../date";
import useProjectTranslation from "../../../translations/useProjectTranslation";

export const Overview = ({project, dataset, group, annotationsStatus, users = []}) => {

    const {translate} = useProjectTranslation();

    const groupEmails = group.userIds.map(userId => {
        const user = users.find(user => user.id === userId);
        if (user == null) {
            return "";
        }
        return user.email;
    })

    return (
        <>
            <ArticleRestitution>
                <HeaderRestitution
                    title={translate('project.project_page.overview.title')}
                />
                <SectionRestitution>
                    <SectionRestitutionRow>
                        <SectionRestitutionColumn>
                            <Restitution label={translate('project.project_page.overview.name')} value={project.name}/>
                            <Restitution label={translate('project.project_page.overview.annotation_type')} value={project.annotationType}/>
                            <Restitution label={translate('project.project_page.overview.creation_date')} value={formatTimestampToString(project.createDate)}/>
                            <Restitution label={translate('project.project_page.overview.nb_files')} value={dataset.files.length}/>
                            <Restitution label={translate('project.project_page.overview.nb_cross_annotation')} value={project.numberCrossAnnotation}/>
                            <Restitution label={translate('project.project_page.overview.annotations_progress')}
                                         value={annotationsStatus ? `${annotationsStatus.percentageNumberAnnotationsDone}%` : '0%'}/>
                        </SectionRestitutionColumn>
                        <SectionRestitutionColumn>
                            <Restitution label={translate('project.project_page.overview.dataset')} value={dataset.name}/>
                            <Restitution label={translate('project.project_page.overview.dataset_type')} value={dataset.type}/>
                            <Restitution label={translate('project.project_page.overview.team')} value={group.name}/>
                            <Restitution label={translate('project.project_page.overview.emails')} values={groupEmails}/>
                        </SectionRestitutionColumn>
                    </SectionRestitutionRow>
                </SectionRestitution>
            </ArticleRestitution>
        </>
    );
};

export default Overview;
