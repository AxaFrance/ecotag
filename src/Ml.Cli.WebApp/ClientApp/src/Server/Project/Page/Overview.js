import React from 'react';
import {
  ArticleRestitution,
  SectionRestitution,
  SectionRestitutionColumn,
  SectionRestitutionRow,
  Restitution,
  HeaderRestitution
} from '@axa-fr/react-toolkit-restitution';
import {formatTimestampToString} from "../../date";

export const Overview = ({ project, dataset, group }) => {
  //const labels = project.labels && project.labels.length > 0 ? project.labels : [];
  //const percentAdvancement = ((100 * (dataset.numberFiles - project.numberTagToDo)) / dataset.numberFiles).toFixed(2);

  return (
    <>
      <ArticleRestitution>
        <HeaderRestitution
            title="Informations générales"
        />
        <SectionRestitution>
          <SectionRestitutionRow >
            <SectionRestitutionColumn>
              <Restitution label="Nom" value={project.name} />
              <Restitution label="Type d'annotation" value={project.typeAnnotation} />
              <Restitution label="Date de création" value={formatTimestampToString(project.createDate)} />
              <Restitution label="Nombre de fichier" value={dataset.files.length} />
              <Restitution label="Nombre annotation croisée" value={project.numberCrossAnnotation} />
              
            </SectionRestitutionColumn>
            <SectionRestitutionColumn>
              <Restitution label="Dataset" value={dataset.name} />
              <Restitution label="Type de dataset" value={dataset.type} />
              <Restitution label="Groupe annotateurs" value={group.name} />
              <Restitution label="Pourcentage d&apos;avancement" value="100%" />
            </SectionRestitutionColumn>
          </SectionRestitutionRow>
        </SectionRestitution>
      </ArticleRestitution>    

    </>
  );
};

export default Overview;
