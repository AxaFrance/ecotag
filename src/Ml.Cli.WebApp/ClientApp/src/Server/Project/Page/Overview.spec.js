import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render, waitFor } from '@testing-library/react';
import {Overview} from './Overview';
import {BrowserRouter as Router} from "react-router-dom";

describe('Overview', () => {
    it('Overview render correctly', async () => {
        const project = {
            "id": "0001",
            "name": "Relevé d'information",
            "datasetId": "0004",
            "numberTagToDo": 10,
            "createDate": new Date("04-04-2011").getTime(),
            "annotationType": "NER",
            annotationStatus:{
                percentageNumberAnnotationsDone:10
            }
        };
        const dataset = {
            "id": "0001",
            "name": "Carte verte",
            "type": "Image",
            "numberFiles": 300,
            "createDate": new Date("10-30-2019").getTime(),
            files: []
        };
        const group = {name: "groupName"};
        const { asFragment, getByText } = render(<Router><Overview dataset={dataset} project={project} group={group}/></Router>);
        const messageEl = await waitFor(() => getByText('Informations générales'));
        expect(messageEl).toHaveTextContent(
            'Informations générales'
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

