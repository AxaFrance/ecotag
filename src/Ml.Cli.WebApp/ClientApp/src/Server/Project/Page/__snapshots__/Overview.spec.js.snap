// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Overview render correctly 1`] = `
<DocumentFragment>
  <h2>
    Informations générales
  </h2>
  <table
    class="af-table"
  >
    <thead
      class="af-table-thead"
    >
      <tr
        class="af-table__tr"
      >
        <th
          class="af-table__th"
        >
          Labels
        </th>
        <th
          class="af-table__th"
        >
          Users
        </th>
        <th
          class="af-table__th"
        >
          Tags restant
        </th>
        <th
          class="af-table__th"
        >
          Pourcentage d'avancement
        </th>
        <th
          class="af-table__th"
        >
          Date de création
        </th>
      </tr>
    </thead>
    <tbody
      class="af-table__body"
    >
      <tr
        class="af-table__tr"
      >
        <td
          class="af-table__cell"
        >
          3
        </td>
        <td
          class="af-table__cell"
        >
          0
        </td>
        <td
          class="af-table__cell"
        >
          10
        </td>
        <td
          class="af-table__cell"
        >
          96.67%
        </td>
        <td
          class="af-table__cell"
        >
          04/04/2011
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
